### PR TITLE
Added navbar #556

### PR DESCRIPTION
**Pull request by AnjaliGarg13**
@Paarmita @divyanshu-rawat 
I have worked upon the dashboard Navbar UI #556. This navbar is not collapsible for now but as suggested by @Paarmita, it is responsive till iPad. 
I have attached the screenshots along. 
Screenshot from 2020-03-22 00-57-10
Screenshot from 2020-03-22 00-57-14
Screenshot from 2020-03-22 00-56-33
Screenshot from 2020-03-22 00-57-45

### CODE REVIEW

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/dashBoardHeader.js [hunk 1]
```diff
@@ -40,64 +53,72 @@ class dashBoardHeader extends React.Component {
     }
     return (
       <React.Fragment>
-        <nav className="navbar navbar-expand-lg whitenav">
+        <nav className="navbar navbar-expand-lg whitenav text-blue">
```
Comment (Paarmita) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/dashBoardHeader.js]: Please remove the bootstrap classes and use emotion style CSS with media queries.

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/dashBoardHeader.js [hunk 2]
```diff
@@ -40,64 +53,72 @@ class dashBoardHeader extends React.Component {
...
+          </div> 
+          
+          
+          <div className="nav-item element">
+            <span>
+              <input  type="text" className="searchbox" value={this.state.text} onChange={this.handleChange}></input>
+            </span>
+          </div>
+
+          <div className="btn-floating mx-1 link contact-btn element envelop" >
```
Comment (Paarmita) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/dashBoardHeader.js]: Remove bootstrap class.

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/dashBoardHeader.js [hunk 3]
```diff
@@ -40,64 +53,72 @@ class dashBoardHeader extends React.Component {
...
+            target="_blank" rel="noopener noreferrer" >
+            <i className="fa fa-envelope fa-button" />
+          </HashLink>
+          </div>
+
+          <div className="btn-floating mx-1 link contact-btn element bell" >
+          <a
+            href="/"
+            target="_blank" rel="noopener noreferrer" >
+            <i className="fa fa-bell fa-button" />
```
Comment (Paarmita) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/dashBoardHeader.js]: Remove fa- fa icons and import SVG. See the latest code.

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Styles/style.scss [hunk 4]
```diff
@@ -0,0 +1,230 @@
+@import './src/ignitus-Shared/styles/ignitus-tokens';
+
+
+
+.nav-item a{
+font-family: Raleway;
```
Comment (Paarmita) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Styles/style.scss]: No need to add font-family as it is already been imported from global CSS.

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Styles/style.scss [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+font-style: normal;
+font-weight: bold;
+text-align: center;
+color: $ignite-color-blue-bg;
+}
+
+.home {
+  position: absolute;
+  left: 8.67vw;
```
Comment (Paarmita) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Styles/style.scss]: Please use rem instead of vh or vw, everwhere.

**Comment by Paarmita** [comment]
On opening inspect in the browser, the navbar is getting misaligned. 

pr

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/dashBoardHeader.js [hunk 6]
```diff
@@ -4,20 +4,33 @@
...
-import { Redirect, Link } from 'react-router-dom';
+import '../Styles/style.scss';
+import ProfileDropDown from '../Components/profileDropdown';
+import { Redirect} from 'react-router-dom';
 import { HashLink } from 'react-router-hash-link';
 import logo from '../../../ignitus-Assets/Images/nav-logo.svg';
 import { withErrorBoundary } from '../../../ignitus-Shared/Components/errorBoundary';
+import StatusDropdown from './statusDropdown';
 
 class dashBoardHeader extends React.Component {
```
Comment (divyanshu-rawat) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/dashBoardHeader.js]: Please don't use class based components, we have already migrated lot fo our `class components ` => `functional components`, here you are extending the base class React.Component and only using `render` method provided by it, and `constructor` as well.

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/dashBoardHeader.js [hunk 7]
```diff
@@ -4,20 +4,33 @@
...
   constructor(props) {
     super(props);
     this.state = {
```
Comment (divyanshu-rawat) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/dashBoardHeader.js]: State declaration can be done using react `useState hook` => https://reactjs.org/docs/hooks-state.html

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/dashBoardHeader.js [hunk 8]
```diff
@@ -4,20 +4,33 @@
...
       redirect: false,
+      text:'',
     };
+    this.handleChange = this.handleChange.bind(this);
```
Comment (divyanshu-rawat) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/dashBoardHeader.js]: If you change it to functional components then you don't need to use `.bind()` anymore, see https://github.com/Ignitus/Ignitus-client/blob/master/src/ignitus-Authentication/ignitus-StudentLogin/Components/Login.jsx#L13 , you can declare function as a constant.

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/dashBoardHeader.js (hunk 1 above)
Comment (divyanshu-rawat) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/dashBoardHeader.js]: Please don't use BS4 as I already mentioned this that we are migration to pure `css` then to => `emotion`, but you can make use of pure `css` as of now, if you don't know `emotion`. see #403

Comment (divyanshu-rawat) [reply src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/dashBoardHeader.js]: And while writing pure css, please write classNames in `camelCase`.

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/dashBoardHeader.js (hunk 3 above)
Comment (divyanshu-rawat) [reply src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/dashBoardHeader.js]: You can use Icons from here, https://github.com/Ignitus/Ignitus-client/tree/master/src/ignitus-Shared/ignitus-DesignSystem/ignitus-Assets/ignitus-Icons, if the icon you want to use, is not there, then create one by downloading `svg` from` font-awesome / material-icon`, and in the end if you still not feel comfortable with the icon components then you can use it for now, but please remove it in the future.

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/profileDropdown.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import '../Styles/style.scss';
+
+
+class ProfileDropdown extends React.Component {
```
Comment (divyanshu-rawat) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/profileDropdown.js]: `Class => functional`

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/profileDropdown.js [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+constructor(props){
+ super(props);
+
+ this.state = {
```
Comment (divyanshu-rawat) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/profileDropdown.js]: `useState hook.`

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/profileDropdown.js [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+       displayMenu: false,
+       User: this.props.User,
+     };
+
+  this.showDropdownMenu = this.showDropdownMenu.bind(this);
+  this.hideDropdownMenu = this.hideDropdownMenu.bind(this);
+
+};
+
+showDropdownMenu(event) {
```
Comment (divyanshu-rawat) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/profileDropdown.js]: declare them as `const` in functional components.

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/profileDropdown.js [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+    event.preventDefault();
+    this.setState({ displayMenu: true }, () => {
+    document.addEventListener('click', this.hideDropdownMenu);
```
Comment (divyanshu-rawat) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/profileDropdown.js]: Please don't use `eventListeners` in the app, we hardly make use of them in react apps theses days they are very bad  `performance` at all,  if you want to have an `hide/show` behaviour of dropdown you can do it with a single variable, I am listing `examples` that will help you understand what I meant, 

https://reactjs.org/docs/handling-events.html
https://blog.logrocket.com/building-a-custom-dropdown-menu-component-for-react-e94f02ced4a1/

Here `sidebar` is written using single variable isExpanded, that toggles the state to hide/show, https://github.com/Ignitus/Ignitus-client/blob/master/src/ignitus-UserInterfaceKit/InterfaceSideNavigation/Components/interfaceSideNavigation.tsx#L31 

Here is the live version of this component. http://www.ignitus.org/interface

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/profileDropdown.js [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+    });
+  }
+
+  hideDropdownMenu() {
+    this.setState({ displayMenu: false }, () => {
+      document.removeEventListener('click', this.hideDropdownMenu);
```
Comment (divyanshu-rawat) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/profileDropdown.js]: 🔝

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/profileDropdown.js [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+    });
+
+  }
+
+  render(props) {
+    return (
+        <div  className="profile-dropdown">
+         <div className="profile-button" onClick={this.showDropdownMenu}> {this.state.User} </div>
+
+         <div class="profile-dropdown-list">
```
Comment (divyanshu-rawat) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/profileDropdown.js]: you can create a separate component for dropdown for eg see here - http://www.ignitus.org/interface/sideNavigation

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/statusDropdown.js [hunk 15]
```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import '../Styles/style.scss';
+
+
+class StatusDropdown extends React.Component {
```
Comment (divyanshu-rawat) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/statusDropdown.js]: `class => functional`

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/statusDropdown.js [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+constructor(){
+ super();
+
+ this.state = {
```
Comment (divyanshu-rawat) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/statusDropdown.js]: `useState`

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/statusDropdown.js [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+
+  this.showDropdownMenu = this.showDropdownMenu.bind(this);
+  this.hideDropdownMenu = this.hideDropdownMenu.bind(this);
+  this.changeStatus = this.changeStatus.bind(this);
+};
+
+showDropdownMenu(event) {
+    event.preventDefault();
+    this.setState({ displayMenu: true }, () => {
+    document.addEventListener('click', this.hideDropdownMenu);
```
Comment (divyanshu-rawat) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Components/statusDropdown.js]: No listeners please.

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Styles/style.scss [hunk 18]
```diff
@@ -0,0 +1,230 @@
...
+.nav-item a{
+font-family: Raleway;
+font-style: normal;
+font-weight: bold;
+text-align: center;
+color: $ignite-color-blue-bg;
+}
+
+.home {
+  position: absolute;
```
Comment (divyanshu-rawat) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Styles/style.scss]: Please avoid using `position absolute` everything can be done using `flexbox`, until and unless you plan to use `z-index`.

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Styles/style.scss [hunk 19]
```diff
@@ -0,0 +1,230 @@
...
+  left: 8.67vw;
+  width: 3.2vw;
+  top: 1.4vh;
+  height: 1.8vh;
```
Comment (divyanshu-rawat) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Styles/style.scss]: Please don't use `vh`, `vw` use `rem` instead.

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Styles/style.scss [hunk 20]
```diff
@@ -0,0 +1,230 @@
...
+.user {
+  position: absolute;
+  left: 80.2vw;
+  width: 8.3vw;
+  height: 8.7vh;
+  top: 2.1vh;
+}
+
+.fa-button{
+  color:$ignite-color-blue-bg !important;
```
Comment (divyanshu-rawat) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Styles/style.scss]: Please don't use `!Important`

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Styles/style.scss [hunk 21]
```diff
@@ -0,0 +1,230 @@
...
+.envelop {
+  position: absolute;
+  left: 72vw;
+  width:3.9vw;
+  height:5.5vh;
+  top: 2vh;
+}
+
+.bell {
+  position: absolute;
```
Comment (divyanshu-rawat) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Styles/style.scss]: No `absolute` positioning.

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Styles/style.scss [hunk 22]
```diff
@@ -0,0 +1,230 @@
...
+.Online {
+  color: red;
+}
+
+.Offline {
+  color: green;
+}
+
+.profile-dropdown {
+  position: relative;
```
Comment (divyanshu-rawat) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Styles/style.scss]: No `relative` positioning.

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Styles/style.scss [hunk 23]
```diff
@@ -0,0 +1,230 @@
...
+.navbar .element {
+  display: inline;
+}
+
+.Online {
+  color: red;
+}
+
+.Offline {
+  color: green;
```
Comment (divyanshu-rawat) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Styles/style.scss]: `scss` variable are defined for `colors`, pick them up else define a new one.

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Styles/style.scss [hunk 24]
```diff
@@ -0,0 +1,230 @@
...
+  padding-left: 1.5vw;
+  padding-top: 0.5vh;
+  border-radius:9px;
+  margin-left: 1vw;
+  font-weight:bold;
+  color:white;
+}
+.profile-button:before{
+  content:"";
+  position:absolute;
```
Comment (divyanshu-rawat) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Styles/style.scss]: No `absolute` positioning.

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Styles/style.scss [hunk 25]
```diff
@@ -0,0 +1,230 @@
...
+  right:1vw;
+  top:1.6vh;
+}
+
+.profile-dropdown-list {  
+  list-style-type: none;
+  margin-top: 2.5vh;
+  padding: 0;
+  top:6.5vh;
+  left:83.6vw !important;
```
Comment (divyanshu-rawat) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Styles/style.scss]: No `important`.

File: src/ignitus-Dashboard/ignitus-Dashboard-Header/Styles/style.scss [hunk 26]
```diff
@@ -0,0 +1,230 @@
...
+  border-radius: 0px 0px 16px 16px;
+}
+.profile-dropdown li:hover {
+  background-color: #000066;
+}
+.profile-dropdown li a {
+  color: $ignite-color-blue-bg !important;
+}
+.profile-dropdown li:hover a{
+  color: #FFF !important;
```
Comment (divyanshu-rawat) [inline src/ignitus-Dashboard/ignitus-Dashboard-Header/Styles/style.scss]: No `important`.

**Comment by divyanshu-rawat** [comment]
@AnjaliGarg13 closing this as of now, you can start it from scratch next time ,or cherry-pick commits from this PR.

**Pull request closed without merging**